build-configuration/pom.xml

```diff
@@ -260,6 +259,9 @@
       <version.org.wildfly.build-tools>1.2.10.Final</version.org.wildfly.build-tools>
       <version.owasp-dependency-check-plugin>4.0.2</version.owasp-dependency-check-plugin>
 
+      <!-- versionx -->
+      <versionx.com.puppycrawl.tools.checkstyle>8.18</versionx.com.puppycrawl.tools.checkstyle>
```

Comment:
Why treat the checkstyle plugin and checkstyle itself differently? They are both needed to build Infinispan, and not needed to run it.